src/sql/src/plan/statement/ddl.rs

```diff
@@ -4971,6 +5019,39 @@ pub fn plan_alter_cluster_rename(
     }
 }
 
+pub fn plan_alter_cluster_swap<F>(
+    scx: &mut StatementContext,
+    name_a: Ident,
+    name_b: Ident,
+    gen_temp_suffix: F,
+) -> Result<Plan, PlanError>
+where
+    F: Fn(&dyn Fn(&str) -> bool) -> Result<String, PlanError>,
+{
+    let cluster_a = scx.resolve_cluster(Some(&name_a))?;
+    let cluster_b = scx.resolve_cluster(Some(&name_b))?;
+
+    let check = |temp_suffix: &str| {
+        let name_temp = Ident::new(format!("cluster_swap_{temp_suffix}"));
+        match scx.catalog.resolve_cluster(Some(name_temp.as_str())) {
+            // Temp name does not exist, so we can use it.
+            Err(CatalogError::UnknownCluster(_)) => true,
+            // Temp name already exists!
+            Ok(_) | Err(_) => false,
```

Comment:
Uncovered, we should test that the handling is correct if the name already exists, but with the random generation that seems hard.

Reply:
Yeah, not sure how to test this since it relies on the random generation